oap-server/server-query-plugin/prometheus-query-plugin/src/main/java/org/apache/skywalking/oap/query/prometheus/handler/PrometheusApiHandler.java

```diff
@@ -0,0 +1,544 @@
+import org.apache.skywalking.oap.query.prometheus.rt.exception.ParseErrorListener;
+import org.apache.skywalking.oap.query.prometheus.rt.result.MatcherSetResult;
+import org.apache.skywalking.oap.query.prometheus.rt.result.MetricsRangeResult;
+import org.apache.skywalking.oap.query.prometheus.rt.PromQLExprQueryVisitor;
+import org.apache.skywalking.oap.query.prometheus.rt.result.ParseResult;
+import org.apache.skywalking.oap.query.prometheus.rt.result.ScalarResult;
+import org.apache.skywalking.oap.server.core.analysis.IDManager;
+import org.apache.skywalking.oap.server.core.analysis.Layer;
+import org.apache.skywalking.oap.server.core.analysis.manual.endpoint.EndpointTraffic;
+import org.apache.skywalking.oap.server.core.analysis.manual.instance.InstanceTraffic;
+import org.apache.skywalking.oap.server.core.analysis.manual.service.ServiceTraffic;
+import org.apache.skywalking.oap.server.core.query.MetricDefinition;
+import org.apache.skywalking.oap.server.core.query.enumeration.Scope;
+import org.apache.skywalking.oap.server.core.query.input.Duration;
+import org.apache.skywalking.oap.server.core.query.type.Endpoint;
+import org.apache.skywalking.oap.server.core.query.type.Service;
+import org.apache.skywalking.oap.server.core.query.type.ServiceInstance;
+import org.apache.skywalking.oap.server.core.storage.annotation.Column;
+import org.apache.skywalking.oap.server.core.storage.annotation.ValueColumnMetadata;
+import org.apache.skywalking.oap.server.library.module.ModuleManager;
+import org.apache.skywalking.oap.server.library.util.StringUtil;
+import org.apache.skywalking.promql.rt.grammar.PromQLLexer;
+import org.apache.skywalking.promql.rt.grammar.PromQLParser;
+
+import static org.apache.skywalking.oap.query.prometheus.rt.PromOpUtils.formatDoubleValue;
+import static org.apache.skywalking.oap.query.prometheus.rt.PromOpUtils.timestamp2Duration;
+
+public class PrometheusApiHandler {
+    private final PrometheusQueryConfig config;
+    private final MetadataQueryV2 metadataQuery;
+    private final MetricsQuery metricsQuery;
+    private final RecordsQuery recordsQuery;
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    public PrometheusApiHandler(final PrometheusQueryConfig config,
+                                ModuleManager moduleManager) {
+        this.config = config;
+        this.metadataQuery = new MetadataQueryV2(moduleManager);
+        this.metricsQuery = new MetricsQuery(moduleManager);
+        this.recordsQuery = new RecordsQuery(moduleManager);
+    }
+
+    @Get
+    @Post
+    @Path("/api/v1/metadata")
+    public HttpResponse metadata(
+        @Param("limit") Optional<Integer> limit,
+        @Param("metric") Optional<String> metric) throws JsonProcessingException {
+        MetadataQueryRsp response = new MetadataQueryRsp();
+        response.setStatus(ResultStatus.success);
+        String regex = "";
+        if (metric.isPresent()) {
+            regex = metric.get();
+        }
```

Comment:
This looks neater, also revise other places where `Optional` appears.

```suggestion
        String regex = metric.orElse("");
```